src/main/sensors/boardalignment.c

```diff
@@ -59,69 +61,55 @@ void initBoardAlignment(const boardAlignment_t *boardAlignment)
     rotationAngles.angles.pitch = degreesToRadians(boardAlignment->pitchDegrees);
     rotationAngles.angles.yaw   = degreesToRadians(boardAlignment->yawDegrees  );
 
-    buildRotationMatrix(&rotationAngles, boardRotation);
+    buildRotationMatrix(&rotationAngles, &boardRotation);
 }
 
-static void alignBoard(float *vec)
+FAST_CODE static void applyRotation(float *vec, fp_rotationMatrix_t *rotationMatrix)
 {
     float x = vec[X];
     float y = vec[Y];
     float z = vec[Z];
 
-    vec[X] = (boardRotation[0][X] * x + boardRotation[1][X] * y + boardRotation[2][X] * z);
-    vec[Y] = (boardRotation[0][Y] * x + boardRotation[1][Y] * y + boardRotation[2][Y] * z);
-    vec[Z] = (boardRotation[0][Z] * x + boardRotation[1][Z] * y + boardRotation[2][Z] * z);
+    vec[X] = (rotationMatrix->m[0][X] * x + rotationMatrix->m[1][X] * y + rotationMatrix->m[2][X] * z);
+    vec[Y] = (rotationMatrix->m[0][Y] * x + rotationMatrix->m[1][Y] * y + rotationMatrix->m[2][Y] * z);
+    vec[Z] = (rotationMatrix->m[0][Z] * x + rotationMatrix->m[1][Z] * y + rotationMatrix->m[2][Z] * z);
+
+}
+
+FAST_CODE static void alignBoard(float *vec)
+{
+    applyRotation(vec, &boardRotation);
+}
+
+FAST_CODE void alignSensor(float *dest, fp_rotationMatrix_t* sensorRotationMatrix)
+{
+    applyRotation(dest, sensorRotationMatrix);
+
+    if (!standardBoardAlignment)
+        alignBoard(dest);
 }
 
+// deprecated
 FAST_CODE void alignSensors(float *dest, uint8_t rotation)
 {
-    const float x = dest[X];
-    const float y = dest[Y];
-    const float z = dest[Z];
-
-    switch (rotation) {
-    default:
-    case CW0_DEG:
-        dest[X] = x;
-        dest[Y] = y;
-        dest[Z] = z;
-        break;
-    case CW90_DEG:
-        dest[X] = y;
-        dest[Y] = -x;
-        dest[Z] = z;
-        break;
-    case CW180_DEG:
-        dest[X] = -x;
-        dest[Y] = -y;
-        dest[Z] = z;
-        break;
-    case CW270_DEG:
-        dest[X] = -y;
-        dest[Y] = x;
-        dest[Z] = z;
-        break;
-    case CW0_DEG_FLIP:
-        dest[X] = -x;
-        dest[Y] = y;
-        dest[Z] = -z;
-        break;
-    case CW90_DEG_FLIP:
-        dest[X] = y;
-        dest[Y] = x;
-        dest[Z] = -z;
-        break;
-    case CW180_DEG_FLIP:
-        dest[X] = x;
-        dest[Y] = -y;
-        dest[Z] = -z;
-        break;
-    case CW270_DEG_FLIP:
-        dest[X] = -y;
-        dest[Y] = -x;
-        dest[Z] = -z;
-        break;
-    }
+    fp_rotationMatrix_t sensorRotationMatrix;
+
+    sensorAlignment_t sensorAlignment;
+
+    buildAlignmentFromRotation(&sensorAlignment, rotation);
+
+#if 0
```

Comment:
Umm... you have ```#if 0``` there.

Reply:
will delete that now, it was temporary code during refactoring process that allowed me to keep the tests running, thanks for spotting it!